trillian/ctfe/config.go

```diff
@@ -210,6 +214,28 @@ func ValidateLogConfig(cfg *configpb.LogConfig) (*ValidatedLogConfig, error) {
 		}
 	}
 
+	switch cfg.ExtraDataIssuanceChainStorageBackend {
+	case configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_CTFE:
+		// Validate the combination of CtfeStorageConnectionString and ExtraDataIssuanceChainStorageBackend
+		if len(cfg.CtfeStorageConnectionString) == 0 {
+			return nil, errors.New("missing ctfe_storage_connection_string when issuance chain storage backend is CTFE")
+		}
+		// Validate CTFEStorageConnectionString
+		if len(cfg.CtfeStorageConnectionString) > 0 {
+			if strings.HasPrefix(cfg.CtfeStorageConnectionString, "mysql") {
+				if _, err := mysql.ParseDSN(strings.Split(cfg.CtfeStorageConnectionString, "://")[1]); err != nil {
+					return nil, errors.New("failed to parse ctfe_storage_connection_string for mysql driver")
+				}
+			} else {
+				return nil, errors.New("unsupported driver in ctfe_storage_connection_string")
+			}
```

Comment:
You can simplify this/improve readability a bit further by flipping the sense around:
```go
if !strings.HasPrefix(..., "mysql") {
  return error
}
if _, err := mysql.ParseDSN(...); err != nil {
  return error
}
```

That'll get rid of the indented `else` block and is a bit more "go idomatic" in the "indent error cases, outdent 'normal' flow" sense.

Reply:
Done.